minetest/minetest
# Add TGA test nodes to devtest

![minetest-mod-devtest-testnodes-tga-use-texture-alpha](https://user-images.githubusercontent.com/60905/150395688-f230e647-288a-4a36-9501-4ab8bbd60c1c.png)

When writing documentation, it came up Minetest does not support all features of all file formats it processes.

In particular, some TGA file types (type 9 & 11) are not supported (and should not be added IMO).

The goal of this PR is to demonstrate Minetest's TGA support and help with debugging.

I have added 10 different nodes for different supported TGA types (1, 2, 3 & 10).

~Note that 16 bit A1R5G5B5 images are not included in this PR. Minetest supports those and one version of the `tga_encoder` mod generates them, but I have found no setting in GIMP to set color depth to 5 bit per color.~

## To do

This PR is Ready for Review.

## How to test

Start devtest and make sure that all test nodes with a texture that reads “TGA” actually render in inventory, hand and world.

Make sure that no TGA-related texture errors occur when loading devtest.

**Edit:** The grayscale TGA test node having a yellowish tint and no transparency is probably a bug in Minetest.

**Edit:** The alpha channel being ignored in the 16bpp textures I added is probably another bug in Minetest.

**Edit:** I have made the node drawtypes glasslike so that they do not have the wallhack appearance.
